File: CountryName/app/src/main/java/com/androidstudiomanju/countryname/MainActivity.java
package com.androidstudiomanju.countryname;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.os.Bundle;
import android.util.Log;

import com.androidstudiomanju.countryname.adapter.CountryAdapter;
import com.androidstudiomanju.countryname.model.CountryModel;
import com.androidstudiomanju.countryname.model.Result;
import com.androidstudiomanju.countryname.service.GetCountryDataService;
import com.androidstudiomanju.countryname.service.RetrofitInstance;

import java.util.ArrayList;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class MainActivity extends AppCompatActivity {
    ArrayList<CountryModel> countries;
    private CountryAdapter countryAdapter;
    private RecyclerView recyclerView;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        getCountries();
    }

    public Object getCountries() {
        GetCountryDataService getCountryDataService = RetrofitInstance.getService();
        Call<Result> call = getCountryDataService.getResult();
        call.enqueue(new Callback<Result>() {
            @Override
            public void onResponse(Call<Result> call, Response<Result> response) {
                Result result = response.body();
                if(result!=null && result.getResult()!=null){
                    countries = (ArrayList<CountryModel>)result.getResult();
                   /* for(CountryModel c : countries){
                        Log.i("TAG",""+c.getName());
                        Log.i("TAG",""+c.getCode());
                        Log.i("TAG",""+c.getStates());
                    }*/
                    ViewData();

                }
            }

            @Override
            public void onFailure(Call<Result> call, Throwable t) {

            }
        });
        return countries;
    }

    private void ViewData() {
        recyclerView = findViewById(R.id.recyclerView);
        countryAdapter = new CountryAdapter(countries);
        RecyclerView.LayoutManager layoutManager = new LinearLayoutManager(MainActivity.this);
        recyclerView.setLayoutManager(layoutManager);
        recyclerView.setAdapter(countryAdapter);

    }
}

File: AndroidKotlin/LifeCycleApp/app/src/main/java/com/androidstudiomanju/lifecycleapp/MainActivity.kt
package com.androidstudiomanju.lifecycleapp

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Toast

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        Toast.makeText(this,"onCreate() is called",Toast.LENGTH_SHORT).show()

        }
    override fun onStart() {
        super.onStart()
        Toast.makeText(this,
            "onStart() is called",
            Toast.LENGTH_SHORT).show()
    }

    override fun onResume() {
        super.onResume()
        Toast.makeText(this,
            "onResume() is called",
            Toast.LENGTH_SHORT).show()
    }

    override fun onRestart() {
        super.onRestart()
        Toast.makeText(this,
            "onRestart() is called",
            Toast.LENGTH_SHORT).show()
    }
    override fun onPause() {
        super.onPause()
        Toast.makeText(this,
            "onPause() is called",
            Toast.LENGTH_SHORT).show()
    }

    override fun onStop() {
        super.onStop()
        Toast.makeText(this,
            "onStop() is called",
            Toast.LENGTH_SHORT).show()
    }

    override fun onDestroy() {
        super.onDestroy()
        Toast.makeText(this,
            "onDestroy() is called",
            Toast.LENGTH_SHORT).show()
    }
}